modulemd/include/private/modulemd-yaml.h

```diff
@@ -692,11 +948,45 @@ skip_unknown_yaml (yaml_parser_t *parser, GError **error);
 
 
 /* A set of macros for simple emitting of common elements */
+/**
+ * NON_EMPTY_TABLE:
+ * @table: A #GHashTable.
+ *
+ * Returns: FALSE if @table is empty, otherwise TRUE.
+ *
+ * Since: 2.0
+ */
 #define NON_EMPTY_TABLE(table) (g_hash_table_size (table) != 0)
 
+/**
+ * NON_EMPTY_ARRAY:
+ * @array: A #GPtrArray.
+ *
+ * Returns: FALSE if @array is empty, otherwise TRUE.
+ *
+ * Since: 2.0
+ */
 #define NON_EMPTY_ARRAY(array) (array->len != 0)
 
-
+/**
+ * EMIT_SCALAR_FULL:
+ * @emitter: (inout): A libyaml emitter object positioned where a scalar
+ * belongs in the YAML document.
+ * @error: (out): A #GError that will return the reason for an output error.
+ * @value: (in): The scalar (string) to be written.
+ * @style: (in): The YAML scalar style for the output.
+ *
+ * Emits scalar @value using style @style.
+ *
+ * DIRECT USE OF THIS MACRO SHOULD BE AVOIDED. This is part of the internal
```

Comment:
This one is actually okay to use directly. It's intended to allow code to specify something other than `YAML_PLAIN_SCALAR_STYLE` (such as the `YAML_FOLDED_SCALAR_STYLE` for the `data.description` section).